[Plane] Vy calc fix

NoMonsters opened the pull request:
Стабильно работает вычисление вертикальной скорости
Надо будет подобрать более хорошие кэфы для фильтров

NoMonsters committed:
Задали период дифф-я

Теперь прериод дифф-я задается напрямую без числа переполненийю.
В прототипе конструктора расписали сами переменные, а не только типы

NoMonsters committed:
Добавили обработку вертикальной скорости

Добавили вызов вычисления вертикальной скорости каждые 10 мС, добавили данные в общий массив данных датчиков

NoMonsters committed:
Изменение вычисления Vy

Теперь вертикальная скорость считается из сырой высоты

NoMonsters committed:
Пофиксили период диф-я

Пока задали напрямую 10 мС, поменяли формат логирования для отладки

NoMonsters committed:
Добавили фильтр высоты

Перед каскадом апериодических звений фильтруем высоты

NoMonsters committed:
Revert "Добавили фильтр высоты"

This reverts commit 780d5cc60a5491a20710706dd7b83a7bb19e60e2.

NoMonsters committed:
Попробовали просто дифференциировать

Не помогло тупое вычитание и деление на dt но пусть будет на всякий

NoMonsters committed:
Передаем объекты сенсоров по ссылке

Раньше передавали по значению из-за этого не созранялись данные между вызовами

NoMonsters committed:
Удалили дифф-тор

NoMonsters committed:
Вернули филтр

Вернули фильтр входной высоты

NoMonsters committed:
Немного подбирали кэфы

Надо будет доподбирать

NoMonsters committed:
Убрал тестовую переменную

PugachA commented on Libraries/Baro/MS5611.cpp:
```diff
@@ -174,7 +174,7 @@ double MS5611::getQFEpressure(void)
-	//calcAltitude(); //uncomment this line and comment the next one if you want to use raw altitude
+	//calcAltitude(); //uncomment this line and comment the next one if you want to use raw altitude, and change filterAltitude to rawAltitude
 	lpAltFilter();
     error = k1*(this->filterAltitude - this->first_filter_output);
```
можно сразу `double error=...`

NoMonsters replied:
пофиксил в двух местах, но в коде их ещё много может быть таких

PugachA commented on Plane/Core/Src/main.cpp:
```diff
@@ -165,14 +166,20 @@ uint8_t Stab(PIReg* reg)
+		ms5611.calcVerticalSpeed();
+		manage_vertical_speed_counter = 0;
+	}
+	data_input[BAROVY] = ms5611.getVerticalSpeed();
```
Нужно ли каждый раз перезаписывать занчение? Наверное лучше внести это в if

NoMonsters replied:
В принципе да, смысла каждый раз писать нет

NoMonsters replied:
пофиксил

PugachA commented on Plane/Core/Src/main.cpp:
```diff
@@ -392,7 +399,8 @@ int main(void)
 		{
-			sprintf(str, "t=%d;mode=%d;omega_x_zad=%d;omega_x=%d;omega_y=%d;omega_z=%d;alt=%d;air_spd=%d", HAL_GetTick(), (int)current_mode ,(int)(0), (int)(data_input[GYROX]*10), (int)(data_input[GYROY]*10), (int)(data_input[GYROZ]*10), (int)(data_input[BARO]*100), data_input[AIR]);
+			//sprintf(str, "t=%d;mode=%d;omega_x_zad=%d;omega_x=%d;omega_y=%d;omega_z=%d;alt=%d;air_spd=%d", HAL_GetTick(), (int)current_mode ,(int)(0), (int)(data_input[GYROX]*10), (int)(data_input[GYROY]*10), (int)(data_input[GYROZ]*10), (int)(data_input[BARO]*100), data_input[AIR]);
+			sprintf(str, "%d, %d\n", (int) (100*data_input[BAROVY]), (int) (100*data_input[BARO]));
```
К сожаления такое не могу пустить в develop, нужно добавить к существующему формату.

NoMonsters replied:
Хорошо, что есть кто это смотрит

NoMonsters replied:
пофиксил

PugachA commented:
Удалите в это PR еще папку со старым проектом Baro

NoMonsters committed:
Перенес запись Vy в условие вычисления

NoMonsters committed:
Вернул логгирование

Вернул старый формат логга для платы ERS. Убрал дебажный кусок, который не используем

NoMonsters committed:
Одновременно объявление и инит переменной

NoMonsters committed:
Удалил проект Baro

NoMonsters committed:
Проинитил массив сенсоров 0

PugachA commented on Plane/Core/Src/main.cpp:
```diff
@@ -392,25 +399,14 @@ int main(void)
 			manage_UART_counter = 0;
 		}
-
-		#ifdef SERVO_DEBUG_UART
```
Аналог этого есть для новой реализации?

NoMonsters replied:
Нет, я это не юзал и никто не юзал из нас. Видимо нам не нужно, поэтому и убрал

PugachA replied:
Мы с Никитой при помощи этого поймали багу с некорректным считыванием ШИМ на холоде

NoMonsters replied:
фигааааа, лан добавил в backlog #143

PugachA commented on Plane/Core/Src/main.cpp:
```diff
@@ -392,25 +399,14 @@ int main(void)
 		{
-			sprintf(str, "t=%d;mode=%d;omega_x_zad=%d;omega_x=%d;omega_y=%d;omega_z=%d;alt=%d;air_spd=%d", HAL_GetTick(), (int)current_mode ,(int)(0), (int)(data_input[GYROX]*10), (int)(data_input[GYROY]*10), (int)(data_input[GYROZ]*10), (int)(data_input[BARO]*100), data_input[AIR]);
+			sprintf(str, "t=%d;mode=%d;omega_x_zad=%d;omega_x=%d;omega_y=%d;omega_z=%d;alt=%d;air_spd=%d;Vy=%d",\
+					HAL_GetTick(), (int)current_mode ,(int)(0),\
```
Нафига `\`?

PugachA replied:
Он сейчас всегда 0 `(int)(0)`?

NoMonsters replied:
это перенос строки, неудобно смотреть в одну строку. Пока не 0, но не хочется пока вытаскивать омега_зад из стабилизации, т.к. она там в локальных переменных

PugachA replied:
Разве без `\` нельзя переносить?

NoMonsters replied:
Хз, а как ещё? Я такое нагуглил, вроде норм решение, кажется нельзя просто так переносить

NoMonsters replied:
вот надо экранировать http://www.prog.org.ru/topic_18709_0.html
